app/uk/gov/hmrc/helptosave/util/WithExponentialBackoffRetry.scala

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosave.util
+
+import akka.actor.{Actor, ActorRef, Cancellable, Scheduler}
+import uk.gov.hmrc.helptosave.util.WithExponentialBackoffRetry.ExponentialBackoffRetry
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
+
+trait WithExponentialBackoffRetry { this: Actor ⇒
```

Comment:
@AndyHWChung  again, unless i miss some requirement , this is an over kill

Reply:
when we retry we don't want to spam the service we're trying to reach - if they're down then a large number of requests to them isn't going to help to situation. see here for some info: https://dzone.com/articles/understanding-retry-pattern-with-exponential-back

Reply:
in this PR, there is no daily scheduled retry at the moment - that will be built in later. The retries in this PR are retries for when the application starts up and there are error retriving a threshold value and errors writing to mongo

Reply:
it would read the value returned by DES in that case - we only try to write to mongo when we've got a value from DES